shotover-proxy/src/message/mod.rs

```diff
@@ -447,27 +451,200 @@ impl Value {
+                values.insert(key, Value::create_element(element));
+            });
+
+            values
+        } else {
+            panic!("Not a UDT. Only CassandraType::Udt should be passed to this method.");
+        }
+    }
+
+    #[allow(clippy::mutable_key_type)]
+    fn create_map(collection: CassandraType) -> BTreeMap<Value, Value> {
+        if let CassandraType::Map(map) = collection {
+            let mut value_list = BTreeMap::new();
+            for (key, value) in map.into_iter() {
+                value_list.insert(Value::create_element(key), Value::create_element(value));
+            }
+
+            value_list
+        } else {
+            panic!("Not a Map. Only CassandraType::Map should be passed to this method");
+        }
+    }
+
+    fn create_element(element: CassandraType) -> Value {
+        match element {
+            CassandraType::Ascii(a) => Value::Ascii(a),
+            CassandraType::Bigint(b) => Value::Integer(b, IntSize::I64),
+            CassandraType::Blob(b) => Value::Bytes(b.into_vec().into()),
+            CassandraType::Boolean(b) => Value::Boolean(b),
+            CassandraType::Counter(c) => Value::Counter(c),
+            CassandraType::Decimal(d) => {
+                let big_decimal = BigDecimal::new(d.unscaled, d.scale.into());
+                Value::Decimal(big_decimal)
+            }
+            CassandraType::Double(d) => Value::Double(d.into()),
+            CassandraType::Float(f) => Value::Float(f.into()),
+            CassandraType::Int(c) => Value::Integer(c as i64, IntSize::I64),
+            CassandraType::Timestamp(t) => Value::Timestamp(t),
+            CassandraType::Uuid(u) => Value::Uuid(u),
+            CassandraType::Varchar(v) => Value::Varchar(v),
+            CassandraType::Varint(v) => Value::Varint(v),
+            CassandraType::Timeuuid(t) => Value::Timeuuid(t),
+            CassandraType::Inet(i) => Value::Inet(i),
+            CassandraType::Date(d) => Value::Date(d),
+            CassandraType::Time(d) => Value::Time(d),
+            CassandraType::Smallint(d) => Value::Integer(d.into(), IntSize::I16),
+            CassandraType::Tinyint(d) => Value::Integer(d.into(), IntSize::I8),
+            CassandraType::List(_) => Value::List(Value::create_list(element)),
+            CassandraType::Map(_) => Value::Map(Value::create_map(element)),
+            CassandraType::Set(_) => Value::Set(Value::create_set(element)),
+            CassandraType::Udt(_) => Value::Udt(Value::create_udt(element)),
+            CassandraType::Tuple(_) => Value::Tuple(Value::create_tuple(element)),
+            CassandraType::Null => Value::NULL,
+        }
+    }
+
+    fn create_list(collection: CassandraType) -> Vec<Value> {
+        match collection {
+            CassandraType::List(collection) => {
+                let mut value_list = Vec::with_capacity(collection.len());
+                for element in collection.into_iter() {
```

Comment:
This should use into_iter -> map -> collect.
Internally that will call Vec::with_capacity and is just as fast as what you have written here.